### ISSUE TITLE
Travis build cannot test extension (missing lib of node-native-keymap)

### ISSUE THREAD

**Issue by buehler** [Microsoft/vscode#27417]
- VSCode Version: 1.12.1
- OS Version: Travis CI Build System

Steps to Reproduce:

1. Write an extension
2. Try to run tests automated (like described on code. https://code.visualstudio.com/docs/extensions/testing-extensions#_running-tests-automatically-on-travis-ci-build-machines)

The problem is, that some native packages are used and those tend to break the build:
https://travis-ci.org/buehler/typescript-hero/builds/237070478#L608

I searched a while now, but I don't find a solution to the problem.

Cheers
Chris

**Comment by sandy081** [comment]
@alexandrudima Assigning to you as the summary mentions `node-native-keymap`. Please un-assign if unrelated.

**Comment by alexandrudima** [comment]
Travis can be configured with the following settings to get a good environment:
https://github.com/Microsoft/vscode/blob/master/.travis.yml

Regardless, in this case, the error has no effect on running the extensions tests. The `native-keymap` node module does not get loaded, but VS Code continues to function correctly without that node module.

TL;DR:
* you can ignore that error message since it does not impact how VS Code runs in a travis build machine (that node module reads the current keyboard layout, etc. which is always standard US on a travis build machine)
* you can configure travis to install those dependencies by taking a look at how we do it in our own travis build file

**Comment by alexandrudima** [comment]
I also believe our recipe for running continuous integration on travis for extensions is terribly outdated. IMHO the recipe should start by adding the MS apt repository and installing vs code from the MS apt repository and executing the extensions tests using a shipped VS Code version. fyi @bpasero @Tyriar 

As a side effet, the apt distribution has its dependencies correctly declared, so the error should never pop up.

**Comment by bpasero** [comment]
@alexandrudima not sure I follow, our extension tests download VS Code from our official release page: https://github.com/Microsoft/vscode-extension-vscode/blob/master/bin/test#L79

We never build VS Code as part of extension tests.

**Comment by alexandrudima** [comment]
@bpasero The test should do `apt add source ...` and `apt install vscode` in order to install all runtime dependencies that vscode needs such that extension authors don't need to spell them all out.

**Comment by bpasero** [comment]
@alexandrudima ah I see, good point.

**Comment by buehler** [comment]
@alexandrudima how would that work?
like this? https://code.visualstudio.com/docs/setup/linux#_installation

**Comment by alexandrudima** [comment]
Yes, I'm thinking that would be the easiest way to install vscode such that it can be used to execute the extension tests. But I'm also a travis noob, so take my suggestions with a grain of salt :)

**Comment by buehler** [comment]
:-) alright, I'm going to try it out.. we'll see. If I got it working, I'm posting the solution here.

**Comment by buehler** [comment]
Alright.. just adding those lines seems to solve the issue.
```yml
addons:
  apt:
    sources:
      - ubuntu-toolchain-r-test
    packages:
      - gcc-4.9
      - g++-4.9
      - gcc-4.9-multilib
      - g++-4.9-multilib
      - zip
      - libgtk2.0-0
      - libx11-dev
      - libxkbfile-dev
```
As soon as I manage to install it via apt get, I'll update this :-)
